docker.go

```diff
@@ -603,20 +607,15 @@ func Generic2DockerOptions(opts ...GenericProviderOption) []DockerProviderOption
 	return converted
 }
 
-// NewDockerProvider creates a Docker provider with the EnvClient
-func NewDockerProvider(provOpts ...DockerProviderOption) (*DockerProvider, error) {
```

Comment:
This could be a breaking change. Let's label de PR for the release notes

Reply:
Ops.. moved below. Nevermind

Reply:
:smile: I tried to not break anything this time :smile: but I was wondering if it makes sense to include `func NewDockerClient() (cli *client.Client, host string, err error)` into the public API or if it would be better to make this private.

The reason I made it public in the first place was to be able to use it in the tests but I guess this shouldn't be an issue anyway because there's no test package?